pynumaflow/batchmapper/_dtypes.py

```diff
@@ -187,8 +190,6 @@ def __iter__(self) -> Iterator[B]:
         return iter(self._responses)
 
     def __getitem__(self, index: int) -> M:
-        if isinstance(index, slice):
-            raise TypeError("Slicing is not supported for BatchResponses")
         return self._responses[index]
```

Comment:
Can we have a quick unit test with slicing and indexing?

Reply:
Sure!